tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -152,8 +141,8 @@ FORCE_INLINE void write_buffers(
         const uint32_t dst_page_index = command_ptr[7];
 
         uint32_t num_to_write;
-        uint32_t src_addr = *reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_rd_ptr_addr(db_buf_switch)) << 4;
-        uint32_t l1_consumer_fifo_limit = src_addr + consumer_cb_size - 1;
+        uint32_t src_addr = get_read_ptr(db_cb_config);
+        uint32_t l1_consumer_fifo_limit = src_addr + (db_cb_config->total_size << 4) - 1;
```

Comment:
I think in general, we make limits be exclusive. Somehow the old inclusive flow stuck around in some cases. We should make it consistent and have it be the exclusive flow.